botorch/acquisition/proximal.py

```diff
@@ -66,29 +68,63 @@ def __init__(
         self.register_buffer("proximal_weights", proximal_weights)
 
         # check model for train_inputs and single batch
-        if not hasattr(self.acq_func.model, "train_inputs"):
+        if not hasattr(model, "train_inputs"):
             raise UnsupportedError(
                 "Acquisition function model must have " "`train_inputs`."
             )
 
-        if (
-            self.acq_func.model.batch_shape != torch.Size([])
-            and self.acq_func.model.train_inputs[0].shape[1] != 1
-        ):
-            raise UnsupportedError(
-                "Proximal acquisition function requires a single batch model"
-            )
+        # change behavior depending on type of model
+        if isinstance(model, ModelListGP):
+            # ModelListGP
+            # make sure that all of the training inputs match and are single batch
+            training_data = model.train_inputs[0][0]
+            for i in range(len(model.train_inputs)):
+
+                if (
+                    self.acq_func.model.batch_shape != torch.Size([])
+                    and model.train_inputs[i][0].shape[1] != 1
+                ):
+                    raise UnsupportedError(
+                        "Proximal acquisition function requires a single batch model"
+                    )
+
+                if not torch.equal(training_data, model.train_inputs[i][0]):
+                    raise UnsupportedError(
+                        "Proximal acquisition function does not support unequal "
+                        "training inputs"
+                    )
+
+            # check to make sure that weights match the training data shape
+            if (
+                len(self.proximal_weights.shape) != 1
+                or self.proximal_weights.shape[0] != training_data.shape[-1]
+            ):
+                raise ValueError(
+                    "`proximal_weights` must be a one dimensional tensor with "
+                    "same feature dimension as model."
+                )
 
-        # check to make sure that weights match the training data shape
-        if (
-            len(self.proximal_weights.shape) != 1
-            or self.proximal_weights.shape[0]
-            != self.acq_func.model.train_inputs[0][-1].shape[-1]
-        ):
-            raise ValueError(
-                "`proximal_weights` must be a one dimensional tensor with "
-                "same feature dimension as model."
-            )
+        else:
+            # SingleTask GP
```

Comment:
This could potentially also be some other model, not necessarily `SingleTaskGP`
```suggestion
            # non-ModelListGP model
```